### DISCUSSION TITLE
The (probably first) TrueType Font renderer for Cosmos

### DISCUSSION THREAD

**Discussion by retrievaaaa** [CosmosOS/Cosmos#2311] [Show and tell]
Hey, I wanted to spread around my new project, CosmosTTF.

**Note: TTF Renderer is based on LunarFonts, which itself is based on a 9 year old C library. Many fonts simply do not work. No workaround as of now.**

If you want to use TTF fonts in your project, go visit
https://github.com/GoldenretriverYT/CosmosTTF

**Comment by quajak** [comment]
Very cool!

**Comment by winbamstudios** [comment]
OMG THIS IS THE THING I NEEDED TYSM FOR MAKING THIS

**Comment by winbamstudios** [comment]
wait a sec TTFManager.RegisterFont isnt working...

**Reply by retrievaaaa** [reply]
What do you mean by it's not working? Any further details?

It's working fine for me.

**Reply by winbamstudios** [reply]
ok hold on i downloaded the code and it has 5 errors but it may be because lunarfonts nuget is installed

**Reply by retrievaaaa** [reply]
Yeah, thats probably why. Most likely colliding with the custom LunarFonts implementation

**Reply by winbamstudios** [reply]
i uninstalled lunarfonts and its still not compiling

**Reply by winbamstudios** [reply]
its doesnt work :(

**Reply by retrievaaaa** [reply]
What errors do you get?

**Reply by winbamstudios** [reply]
The type or namespace name 'List<>' could not be found
'GlyphResult': cannot have instance property or field initializers in structs
'GlyphResult': cannot have instance property or field initializers in structs
The type or namespace name 'Exception' could not be found
The type or namespace name 'Exception' could not be found

**Reply by retrievaaaa** [reply]
Please check if your projects are targetting .NET 6.0 and are on the latest DevKit

**Reply by winbamstudios** [reply]
oh im using userkit with .net 5.0

**Reply by winbamstudios** [reply]
can u backport it tho pls

**Reply by retrievaaaa** [reply]
Sorry, but I wont. Your best bet is to add
```
using System;
using System.Collections;
using System.Collections.Generic;
```

to the top of TTFManager.cs
and changing
```
public Bitmap bmp;
public int offY = 0;
public int offX = 0;
```

in the GlyphResult struct to
```
public Bitmap bmp;
public int offY;
public int offX;
```

Then it should in theory compile, but I can't gurantee its gonna work properly. You should be using the devkit anyway.
This would be the final TTFManager.cs after the edits: https://gist.github.com/GoldenretriverYT/5c1dfbcf2aa1560f9cad7d46536cbeea

**Reply by winbamstudios** [reply]
ok

**Comment by winbamstudios** [comment]
UPDATE: i'm using devkit now time for test number 2

**Reply by retrievaaaa** [reply]
did it work?

**Reply by winbamstudios** [reply]
y e s

**Reply by retrievaaaa** [reply]
awesome!

**Comment by ghost** [comment]
Looks nice.

**Comment by winbamstudios** [comment]
hello im here again
I still use this library and i love it

**Reply by SuperWasTaken** [reply]
How do you bulid the TTF library?
